cmd/baton-okta/main.go

```diff
@@ -39,8 +39,16 @@ func main() {
 
 func getConnector(ctx context.Context, cfg *config) (types.ConnectorServer, error) {
 	l := ctxzap.Extract(ctx)
-
-	cb, err := connector.New(ctx, cfg.Domain, cfg.ApiToken, cfg.SyncInactiveApps)
+	config := map[string]any{
+		"Domain":           cfg.Domain,
+		"ApiToken":         cfg.ApiToken,
+		"SyncInactiveApps": cfg.SyncInactiveApps,
+		"OktaClientId":     cfg.OktaClientId,
+		"OktaPrivateKey":   cfg.OktaPrivateKey,
+		"OktaPrivateKeyId": cfg.OktaPrivateKeyId,
+		"OktaProvisioning": cfg.OktaProvisioning,
+	}
+	cb, err := connector.New(ctx, config)
```

Comment:
Instead of creating a new config struct, why not just pass in cfg? That would save a lot of checking and casting in the New() function.